src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -421,7 +421,7 @@ public bool ContainsDiagnostics
         /// </summary>
         public bool Contains(SyntaxNode node)
         {
-            if (node == null || !node.FullSpan.IntersectsWith(node.FullSpan))
+            if (node == null || !this.FullSpan.IntersectsWith(node.FullSpan))
```

Comment:
From our discussion, I will leave as-is, as the current implementation may be more performant. 
It only partially walks up one tree, while calling `SyntaxTree` walks all the way up the tree (when that property wasn't cached) and we do it twice.